server/apps/publish/published_item.py

```diff
@@ -174,24 +293,162 @@ def insert_into_text_archive(self, doc):
-                        format(doc['item_id'], doc.get('headline'), doc.get('_version'), doc.get('expiry')))
+            text_archive_service.delete_action({config.ID_FIELD: doc[config.ID_FIELD]})
+            logger.info('Deleted published item {} with headline {} and version {} and expiry {} '
+                        'as the state of an article is killed.'.format(doc['item_id'], doc.get('headline'),
+                                                                       doc.get('_version'), doc.get('expiry')))
 
-    def find_one(self, req, **lookup):
-        item = super().find_one(req, **lookup)
-        handle_existing_data(item)
+    def __upsert_into_legal_archive(self, doc):
+        """
+        Checks if the article is available in legal archive. If available updates the article otherwise inserts.
+        """
 
-        return item
+        legal_archive_doc = doc.copy()
+        logging.info('Preparing Article to be inserted into Legal Archive %s' % legal_archive_doc.get('unique_name'))
 
-    def __set_published_item_expiry(self, doc):
-        desk_id = doc.get('task', {}).get('desk', None)
-        desk = {}
-        if desk_id:
-            desk = superdesk.get_resource_service('desks').find_one(req=None, _id=desk_id)
-        expiry_minutes = desk.get('published_item_expiry', config.PUBLISHED_ITEMS_EXPIRY_MINUTES)
-        doc['expiry'] = get_expiry_date(expiry_minutes)
+        # Removing Irrelevant properties
+        legal_archive_doc[config.ID_FIELD] = legal_archive_doc['item_id']
+        del legal_archive_doc[config.ETAG]
+        del legal_archive_doc['item_id']
+
+        logging.info('Removed Irrelevant properties from the article %s' % legal_archive_doc.get('unique_name'))
+
+        # Fetch Formatted Items
+        lookup = {'item_id': legal_archive_doc[config.ID_FIELD], 'item_version': legal_archive_doc[config.VERSION]}
+        formatted_items = list(get_resource_service('formatted_item').get(req=None, lookup=lookup))
+        assert len(formatted_items) > 0, \
+            "Formatted Item(s) are empty for a published item %s" % legal_archive_doc[config.ID_FIELD]
+        logging.info('Fetched formatted items for the article %s' % legal_archive_doc.get('unique_name'))
+
+        # Fetch Publish Queue Items
+        formatted_item_ids = [str(formatted_item[config.ID_FIELD]) for formatted_item in formatted_items]
+        query = {'$and': [{'formatted_item_id': {'$in': formatted_item_ids}}]}
+        queue_items = list(get_resource_service('publish_queue').get(req=None, lookup=query))
+        assert len(queue_items) > 0, \
+            "Transmission Details are empty for published item %s" % legal_archive_doc[config.ID_FIELD]
+        logging.info('Fetched transmission details for article %s' % legal_archive_doc.get('unique_name'))
+
+        output_channel_ids = list(set([str(queue_item['output_channel_id']) for queue_item in queue_items]))
```

Comment:
We can use set comprehension here directly, avoiding the need for the "inner" list.